src/code/views/node-view.coffee

```diff
@@ -16,6 +16,11 @@ NodeTitle = React.createFactory React.createClass
     if @props.isEditing
       @inputElm().off()
 
+  componentWillUpdate: (nextProps) ->
+    # mark the title as updated even if no change was made when it leaves edit mode
+    @props.node.titleUpdated = true if @props.isEditing and not nextProps.isEditing
+    @canDeleteWhenEmpty = @props.node.addedThisSession and not @props.node.titleUpdated
```

Comment:
You are right - when I added it I thought that the input was being mounted/unmounted whenever the node title was edited.  I see now that it always remains mounted so I will move that to a view variable.